streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -397,6 +421,17 @@ public V apply(final K aggKey, final V aggOne, final V aggTwo) {
         return aggregate(initializer, aggregator, sessionMerger, sessionWindows, valSerde, storeSupplier);
     }
 
+
+    private <T> KTable<K, T> doAggregate(final KStreamAggProcessorSupplier<K, ?, V, T> aggregateSupplier,
+                                         final String functionName,
+                                         final MaterializedInternal<K, T, KeyValueStore<Bytes, byte[]>> materializedInternal) {
+
+        final StoreBuilder<KeyValueStore<K, T>> storeBuilder = new KeyValueStoreMaterializer<>(materializedInternal)
+                .materialize();
+        return aggregateBuilder.build(aggregateSupplier, functionName, storeBuilder, true);
```

Comment:
This is related to the above comment: Should `isQuerable` always be true? If we have overloads without `Materialized`, and we use a default materialization method, in that case it should not be queryable?

Reply:
in this case `isQueryable` is true when  `Materialized` is provided, which is not always the case, i.e, we have:
```
aggregate(final Initializer<VR> initializer,
                                 final Aggregator<? super K, ? super V, VR> aggregator,
                                 final Serde<VR> aggValueSerde)
```
which is not queryable.